File: 백준/C언어/나머지.c
// 3052
// 나머지
// C언어
// 첫째 줄에, 42로 나누었을 때, 서로 다른 나머지가 몇 개 있는지 출력하시오.

#include <stdio.h>

int main(void)
{
	int num, count = 0, value = 0;
	int remain[10];

	for (int i = 0; i < 10; i++)
	{
		scanf("%d", &num);
		remain[i] = num % 42;
	}
	for (int j = 0; j < 10; j++)
	{
		for (int k = 0; k < j; k++)
		{
			if (remain[j] == remain[k])
				count++;
		}
		if (count == 0)
			value++;
	}
	printf("%d", value);

	return 0;
}

File: 백준/파이썬/1차원 배열/나머지.py
# 3052
# 나머지
# Python
# 첫째 줄에, 42로 나누었을 때, 서로 다른 나머지가 몇 개 있는지 출력하시오.

remain = []

for i in range(10):
    num = int(input())
    remain.append(num % 42)
remain = set(remain)
print(len(remain))
